code/game/machinery/camera/camera_assembly.dm

```diff
@@ -31,6 +31,10 @@
 	if(building)
 		setDir(ndir)
 
+/obj/structure/camera_assembly/Destroy()
+	upgrades = null

```

Comment:
You should delete the upgrades in the list first, then `Cut()` the list, instead of just nulling it out.

Even if